alma/controllers/hook/DisplayPaymentReturnHookController.php

```diff
@@ -43,16 +44,23 @@ public function run($params)
     {
         $this->context->controller->addCSS($this->module->_path . 'views/css/alma.css', 'all');
 
+        $payment = null;
         $order = array_key_exists('objOrder', $params) ? $params['objOrder'] : $params['order'];
         $orderPayment = $this->getOrderPaymentOrFail($order);
-        $alma = ClientHelper::defaultInstance();
         $almaPaymentId = $orderPayment->transaction_id;
-
-        try {
-            $payment = $alma->payments->fetch($almaPaymentId);
-        } catch (RequestError $e) {
-            Logger::instance()->error("[Alma] Error fetching payment with ID {$almaPaymentId}: {$e->getMessage()}");
-            throw new PaymentValidationError(null, $e->getMessage());
+        if (!$almaPaymentId) {
+            $msg = '[Alma] Payment_id not found';
+            Logger::instance()->error($msg);
+            throw new RenderPaymentException($msg);
+        }
+        $alma = ClientHelper::defaultInstance();
+        if ($alma) {
+            try {
+                $payment = $alma->payments->fetch($almaPaymentId);
+            } catch (RequestError $e) {
+                Logger::instance()->error("[Alma] Error fetching payment with ID {$almaPaymentId}: {$e->getMessage()}");
+                throw new PaymentValidationError(null, $e->getMessage());
```

Comment:
This is not a `ValidationError` ... as we said before, validation is done before this step...

Here, the controller just try to render something... if controller can't fetch a payment, it just fail to render what he want.

I suggest you to `throw new RenderPaymentException` here too and catch only this exception in `alma.php` main controller 😉